engine/engine.go

```diff
@@ -543,7 +554,7 @@ func runAvailableSteps(dbp zesty.DBProvider, modifiedSteps map[string]bool, res
 				}
 				// rebuild step dependency tree to include generated loop steps
 				res.BuildStepTree()
-				commit(dbp, res, t)
+				commit(dbp, res, nil)
```

Comment:
I assume there's a reason to not pass t here?

Reply:
As seen with @rbeuque74 , the task does not need to be commited at this point.

Reply:
Ok, that's what I expected but I havent rechecked the flow